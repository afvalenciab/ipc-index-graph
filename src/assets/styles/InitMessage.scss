.init__container{
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: 10px 50px;

  .init__description {
    margin: 0;
    text-align: justify;
    font-size: 30px;
    width: 60%;

    span {
      color: #006AFA;
      font-weight: 700;
    }
  }

  .init__invitation {
    letter-spacing: 1.5px;
    font-weight: 900;
    font-size: 18px;
    margin: 20px;
  }

  button {
    border: 0;
    outline: none;
    padding: 15px 60px;
    font-size: 16px;
    background: #006AFA;
    color: #fff;
    cursor: pointer;
    border-radius: 25px;
    letter-spacing: 3px;
    transition: .3s;
    will-change: transform;
  }

  button:hover {
    transform: translateY(-5px);
  }
}