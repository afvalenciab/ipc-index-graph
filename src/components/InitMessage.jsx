import React from 'react';
import { useHistory } from 'react-router-dom';
import '../assets/styles/InitMessage.scss';

const InitMessage = (props) => {
  const history = useHistory();
  const handleClick = () => {
    history.push('/login');
  };

  return (
    <section className='init__container'>
      <p className='init__description'>
        El Índice de Precios y Cotizaciones
        <span> (IPC) </span>
        es el indicador oficial de la
        evolución del mercado accionario, ya que busca medir el rendimiento de
        las acciones de mayor tamaño y liquidez listadas en la Bolsa Mexicana de
        Valores.
      </p>
      <p className='init__invitation'>Ingresa y conoce el comportamiento del índice</p>
      <button type='button' onClick={handleClick}>Ingresar</button>
    </section>
  );
};

export default InitMessage;
