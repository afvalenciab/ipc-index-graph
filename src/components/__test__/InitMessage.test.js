import React from 'react';
import { create } from 'react-test-renderer';
import ProviderMock from '../../__mocks__/ProviderMocks';
import InitMessage from '../InitMessage';

describe('InitMessage testing', () => {
  test('Match Snapshot', () => {
    const initMessage = create(
      <ProviderMock>
        <InitMessage />
      </ProviderMock>
    );
    expect(initMessage.toJSON()).toMatchSnapshot();
  });
});
