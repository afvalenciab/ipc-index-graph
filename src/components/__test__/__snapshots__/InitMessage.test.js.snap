// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InitMessage testing Match Snapshot 1`] = `
<section
  className="init__container"
>
  <p
    className="init__description"
  >
    El Índice de Precios y Cotizaciones
    <span>
       (IPC) 
    </span>
    es el indicador oficial de la evolución del mercado accionario, ya que busca medir el rendimiento de las acciones de mayor tamaño y liquidez listadas en la Bolsa Mexicana de Valores.
  </p>
  <p
    className="init__invitation"
  >
    Ingresa y conoce el comportamiento del índice
  </p>
  <button
    onClick={[Function]}
    type="button"
  >
    Ingresar
  </button>
</section>
`;
